src/Features/Core/Portable/CodeStyle/AbstractCodeStyleDiagnosticAnalyzer.cs

```diff
@@ -57,15 +58,23 @@ internal abstract class AbstractCodeStyleDiagnosticAnalyzer : DiagnosticAnalyzer
                 Descriptor, UnnecessaryWithoutSuggestionDescriptor, UnnecessaryWithSuggestionDescriptor);
         }
 
+        protected AbstractCodeStyleDiagnosticAnalyzer(ImmutableArray<DiagnosticDescriptor> diagnosticDescriptors)
+        {
+            Debug.Assert(diagnosticDescriptors.Length > 0);
+            SupportedDiagnostics = diagnosticDescriptors;
+        }
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; }
+
         protected DiagnosticDescriptor CreateUnnecessaryDescriptor()
             => CreateUnnecessaryDescriptor(DescriptorId);
 
         protected DiagnosticDescriptor CreateUnnecessaryDescriptor(string descriptorId)
-            => CreateDescriptorWithId(
-                descriptorId, _localizableTitle, _localizableMessageFormat,
-                DiagnosticCustomTags.Unnecessary);
+            => CreateUnnecessaryDescriptor(descriptorId, _localizableTitle, _localizableMessageFormat, _configurable);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; }
+        protected static DiagnosticDescriptor CreateUnnecessaryDescriptor(
+            string id, LocalizableString title, LocalizableString messageFormat, bool configurable, bool enabledByDefault = true)
+            => CreateDescriptor(id, title, messageFormat, configurable, enabledByDefault, DiagnosticCustomTags.Unnecessary);
```

Comment:
can you not use default parameters in internal members?  It's good to force code to have to think about what the right value is here.